lib/new_relic/harvest/telemetry_sdk/dimensional_metrics/harvester.ex

```diff
@@ -0,0 +1,165 @@
+defmodule NewRelic.Harvest.TelemetrySdk.DimensionalMetrics.Harvester do
+  use GenServer
+
+  @moduledoc false
+
+  alias NewRelic.Harvest
+  alias NewRelic.Harvest.TelemetrySdk
+
+  @interval_ms TelemetrySdk.Config.lookup(:dimensional_metrics_harvest_cycle)
+
+  @valid_types [:count, :gauge, :summary]
+
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, [])
+  end
+
+  def init(_) do
+    {:ok,
+     %{
+       start_time_ms: System.system_time(:millisecond),
+       metrics: %{}
+     }}
+  end
+
+  # API
+
+  @spec report_dimensional_metric(:count | :gauge | :summary, atom() | binary(), any, map()) ::
+          :ok
+  def report_dimensional_metric(type, name, value, attributes) when type in @valid_types do
```

Comment:
I've been burned more than once by the fact that attribute values can't be null in the dimensional metric API.

> _ Values can be strings, JSON numbers, or booleans_

I'm wondering if we should scan attribute values for nil and return `{:error, ...}` if we detect it.

The metric slice will get dropped and users won't know the issue is there unless they check the `NrIntegrationEvent`